cli/src/main.rs

```diff
@@ -21,32 +22,145 @@ fn main() {
+                .about("Access lists of transactions")
+                .setting(AppSettings::SubcommandRequired)
+                .subcommand(SubCommand::with_name("all").about("Fetch all transactions")),
+        )
+}
 
-    let client = Client::horizon_test().unwrap();
-    match matches.subcommand() {
-        ("account", Some(sub_m)) => match sub_m.subcommand() {
-            ("details", Some(sub_m)) => {
-                let id = sub_m.value_of("ID").expect("ID is required");
-                let endpoint = account::Details::new(id);
-                if let Ok(account) = client.request(endpoint) {
-                    println!("ID:       {}", account.id());
-                    println!("Sequence: {}", account.sequence());
-                } else {
-                    eprintln!("Could not find account with id {}", id);
-                    ::std::process::exit(1);
+fn print_help_and_exit() {
+    build_app().print_help().expect("Error printing help");
+    println!();
+    ::std::process::exit(1);
+}
+
+fn next_page() -> bool {
+    println!("-- press q to quit --");
+    let mut input = String::new();
+    match ::std::io::stdin().read_line(&mut input) {
+        Ok(_) => !input.starts_with("q"),
+        _ => false,
+    }
+}
+
+mod account {
+    use clap::ArgMatches;
+    use stellar_client::{endpoint::{account, Order}, error::Error, sync::Client};
+    use super::next_page;
+
+    pub fn details<'a>(client: Client, matches: &'a ArgMatches) {
+        let id = matches.value_of("ID").expect("ID is required");
+        let endpoint = account::Details::new(id);
+        match client.request(endpoint) {
+            Ok(account) => {
+                println!("ID:       {}", account.id());
+                println!("Sequence: {}", account.sequence());
+            }
+            Err(Error::BadResponse(err)) => {
+                eprintln!("{}", err);
+                ::std::process::exit(1);
+            }
+            err => {
+                eprintln!("An unknown error occurred: {:?}", err);
+                ::std::process::exit(1);
+            }
+        }
+    }
+
+    pub fn transactions<'a>(client: Client, matches: &'a ArgMatches, cursor: Option<String>) {
+        let id = matches.value_of("ID").expect("ID is required");
+        let mut endpoint = account::Transactions::new(id).order(Order::Desc);
+
+        if let Some(c) = cursor {
+            endpoint = endpoint.cursor(&c);
+        }
+
+        match client.request(endpoint) {
+            Ok(records) => {
+                for txn in records.records().iter() {
+                    println!("ID:         {}", txn.id());
+                    println!("account id: {}", txn.source_account());
+                    println!("created at: {}", txn.created_at());
+                    println!("");
+                }
+                if next_page() {
+                    transactions(client, &matches, Some(records.next_cursor().to_string()))
```

Comment:
🕺

Interesting that there is always a next_cursor() value, but the API will send back a 400 if you call the `_links#next` url that horizon returns.  Nothing to do with your code, just writing down thoughts.

Reply:
Interesting, good point. It ends up just returning empty sets. I should break the loop when there's nothing.